src/xdg.c

```diff
@@ -175,17 +175,33 @@ handle_request_maximize(struct wl_listener *listener, void *data)
 		/*store_natural_geometry*/ true);
 }
 
+static void
+set_fullscreen_from_request(struct view *view,
+		struct wlr_xdg_toplevel_requested *requested)
+{
+	if (!view->fullscreen && requested->fullscreen
+			&& requested->fullscreen_output) {
+		struct output *output = output_from_wlr_output(view->server,
+			requested->fullscreen_output);
+		if (output_is_usable(output)) {
+			view->output = output;
+		} else {
+			wlr_log(WLR_ERROR,
+				"invalid output in fullscreen request");
+		}
```

Comment:
I am not sure right now but I think clients can ask for fullscreen without supplying an output.
In the case of `!view->been_mapped` that should likely be the cursor output, otherwise the output the view is mainly on.

Reply:
That should be handled all right; view_set_fullscreen() is called regardless and will use the current view->output in that case.